cmd/podman/images/scan.go

```diff
@@ -0,0 +1,158 @@
+
+	flagName = "env"
+	flags.StringSliceVarP(
+		&scanOptions.Env,
+		flagName, "e", nil,
+		"Set environment variables in scanner tool container",
+	)
+	_ = cmd.RegisterFlagCompletionFunc(flagName, completion.AutocompleteNone)
+
+}
+
+func init() {
+	registry.Commands = append(registry.Commands, registry.CliCommand{
+		Mode:    []entities.EngineMode{entities.ABIMode},
+		Command: imageScanCommand,
+		Parent:  imageCmd,
+	})
+	scanFlags(imageScanCommand)
+}
+
+func scan(cmd *cobra.Command, args []string) error {
+	imageToScan := args[0]
+
+	// render all scanner args with the mount-point (if necessary)
+	type ScanTemplate struct {
+		MountPoint string
+	}
+	values := ScanTemplate{MountPoint: scanOptions.MountPoint}
+	tpl := template.New("podman-scan")
+
+	scannerArgs := append([]string{scanOptions.ScannerImage}, args[1:]...)
+	for i, arg := range scannerArgs {
+		parsedTpl, err := tpl.Parse(arg)
+		if err != nil {
+			continue
+		}
+
+		buf := bytes.NewBufferString("")
+		if err = parsedTpl.Execute(buf, values); err != nil {
+			continue
+		}
+		scannerArgs[i] = buf.String()
+	}
+
+	mountOptions := fmt.Sprintf("type=image,src=%s,target=%s,readwrite=false", imageToScan, scanOptions.MountPoint)
+
+	ctrConfig := registry.PodmanConfig()
+
+	ctrOpts := common.ContainerCLIOpts{
+		Mount:             []string{mountOptions},
+		ImageVolume:       common.DefaultImageVolume,
+		HealthInterval:    common.DefaultHealthCheckInterval,
+		HealthRetries:     common.DefaultHealthCheckRetries,
+		HealthStartPeriod: common.DefaultHealthCheckStartPeriod,
+		HealthTimeout:     common.DefaultHealthCheckTimeout,
+		HTTPProxy:         true,
+		MemorySwappiness:  -1,
+		// TODO: scanners may need these options to be made available to the user, yes?
+		Net:           &entities.NetOptions{},
+		CGroupsMode:   ctrConfig.Cgroups(),
+		Devices:       ctrConfig.Devices(),
+		Env:           scanOptions.Env,
+		Ulimit:        ctrConfig.Ulimits(),
+		InitPath:      ctrConfig.InitPath(),
+		Pull:          ctrConfig.Engine.PullPolicy,
+		SdNotifyMode:  define.SdNotifyModeContainer,
+		ShmSize:       ctrConfig.ShmSize(),
+		StopTimeout:   ctrConfig.Engine.StopTimeout,
+		Systemd:       "true",
+		Timezone:      ctrConfig.TZ(),
+		Umask:         ctrConfig.Umask(),
+		UserNS:        os.Getenv("PODMAN_USERNS"),
+		Volume:        ctrConfig.Volumes(),
+		SeccompPolicy: "default",
+	}
+
+	s := specgen.NewSpecGenerator(scanOptions.ScannerImage, false)
+	if err := common.FillOutSpecGen(s, &ctrOpts, scannerArgs); err != nil {
+		return err
+	}
+
+	s.RawImageName = scanOptions.ScannerImage
+	s.Terminal = true
+
+	runOpts := entities.ContainerRunOptions{
+		OutputStream: os.Stdout,
+		ErrorStream:  os.Stderr,
+		InputStream:  os.Stdin,
+		Rm:           true,
+		Spec:         s,
+		SigProxy:     true,
+		DetachKeys:   ctrConfig.DetachKeys(),
+	}
+
+	report, err := registry.ContainerEngine().ContainerRun(registry.GetContext(), runOpts)
```

Comment:
I might be misunderstanding what this does, but why is running a container w/ the image necessary? It looks like `docker scan` just gives you information about the image?